modules/optableBidAdapter.js

```diff
@@ -17,17 +17,42 @@ const BIDDER_CODE = 'optable';
 const DEFAULT_REGION = 'ca'
 const DEFAULT_ORIGIN = 'https://ads.optable.co'
 
+function getOrigin() {
+  return config.getConfig('optable.origin') ?? DEFAULT_ORIGIN;
+}
+
+function getBaseUrl() {
+  const region = config.getConfig('optable.region') ?? DEFAULT_REGION;
+  return `${getOrigin()}/${region}`
+}
+
 export const spec = {
   code: BIDDER_CODE,
   isBidRequestValid: function(bid) { return !!bid.params?.site },
   buildRequests: function(bidRequests, bidderRequest) {
-    const region = config.getConfig('optable.region') ?? DEFAULT_REGION
-    const origin = config.getConfig('optable.origin') ?? DEFAULT_ORIGIN
-    const requestURL = `${origin}/${region}/ortb2/v1/ssp/bid`
+    const requestURL = `${getBaseUrl()}/ortb2/v1/ssp/bid`
     const data = converter.toORTB({ bidRequests, bidderRequest, context: { mediaType: BANNER } });
-
     return { method: 'POST', url: requestURL, data }
   },
+  buildPAAPIConfigs: function(bidRequests) {
+    const origin = getOrigin();
+    return bidRequests
+      .filter(req => req.ortb2Imp?.ext?.ae)
+      .map(bid => ({
+        bidId: bid.bidId,
+        config: {
+          seller: origin,
+          decisionLogicURL: `${getBaseUrl()}/paapi/v1/ssp/decision-logic.js?origin=${bid.params.site}`,
+          interestGroupBuyers: [origin],
```

Comment:
Is there any consideration to have the `PAAPI` configs be dynamically configured, possibly from storing into localstorage from a previous auction result. I'm thinking about origins specifically but could be other things we might want to be dynamic. As a SSP the buyer origin is not a hardcoded value as we have multiple buyers depending on context.

Reply:
> Is there any consideration to have the PAAPI configs be dynamically configured, possibly from storing into localstorage from a previous auction result.

The interface should be general enough to allow this from an adapter. If it becomes a common pattern we can consider extracting it to a shared library - right now I don't know how that would look like (I expect SSP will still want to get in the first auction, which requires hardcoded buyers).

>  As a SSP the buyer origin is not a hardcoded value as we have multiple buyers depending on context.

Yes, and IMO this will quickly devolve into every prebid bundle containing every buyer that exists - but alas we decided that's what we want.